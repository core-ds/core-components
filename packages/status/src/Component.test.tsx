import React from 'react';
import { render } from '@testing-library/react';

import { Status, colors } from './index';

describe('Status', () => {
    describe('Snapshots tests', () => {
        it.each(['muted', 'contrast', 'muted-alt'] as const)(
            'should match view="%s" snapshot',
            (view) => {
                expect(render(<Status view={view}>Label</Status>).container).toMatchSnapshot();
            },
        );
    });

    it('should set `data-test-id` attribute', () => {
        const dataTestId = 'test-id';
        const { getByTestId } = render(<Status dataTestId={dataTestId} />);

        expect(getByTestId(dataTestId).tagName).toBe('SPAN');
    });

    describe('Classes tests', () => {
        it('should set `className` class', () => {
            const className = 'test-class';
            const { container } = render(<Status className={className} />);

            expect(container.firstElementChild).toHaveClass(className);
        });

        it('should set `view=muted-alt, color=green` by default', () => {
            const { container } = render(<Status />);

            expect(container.firstElementChild).toHaveClass('muted-alt');
            expect(container.firstElementChild).toHaveClass('green');
        });

        it.each(['muted-alt', 'contrast', 'muted'] as const)('should set view="%s"', (view) => {
            const { container } = render(<Status view={view}>Label</Status>);

            expect(container.firstElementChild).toHaveClass(view);
        });

        it.each(colors)('should set color="%s"', (color) => {
            const { container } = render(<Status color={color}>Label</Status>);

            expect(container.firstElementChild).toHaveClass(color);
        });
    });

    it('should unmount without errors', () => {
        const { unmount } = render(<Status />);

        expect(unmount).not.toThrowError();
    });
});
