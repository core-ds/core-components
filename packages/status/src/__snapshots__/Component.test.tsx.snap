// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Snapshots tests should match view="contrast" snapshot 1`] = `
<div>
  <span
    class="component green contrast"
  >
    <span
      class="ellipsis"
    >
      Label
    </span>
  </span>
</div>
`;

exports[`Status Snapshots tests should match view="soft" snapshot 1`] = `
<div>
  <span
    class="component green soft"
  >
    <span
      class="ellipsis"
    >
      Label
    </span>
  </span>
</div>
`;
